trin-core/src/portalnet/protocol.rs

```diff
@@ -309,103 +305,3 @@ impl PortalnetEvents {
         }
     }
 }
-
-impl PortalnetProtocol {
```

Comment:
Yes, the idea for the overlay protocol is that it will handle all requests from the overlay networks (state, history) and dispatch them to the discv5 protocol TalkReq.
So the implication is that every network should implement the overlay protocol and the overlay protocol is where we can encapsulate the logic for handling common network requests/responses. 

If there is something specific to a network that we don't want to implement it in the overlay protocol, we can encapsulate it on the network level aka `struct StateProtocol`, `struct HistoryProtocol`. Probably it makes sense to differentiate between 'protocols' and 'networks' and rename those struct to `StateNetwork`, `HistoryNetwork` for example...

Reply:
That sounds like a great idea, if you don't mind tackling it. I'd also recommend a quick comment above the struct definitions giving a brief explanation of the data types and their responsibilities